### PR TITLE
add top level documentation for compile

### PR DESCRIPTION
**Pull request by daniel-makerx**

### CODE REVIEW

File: docs/lg-compile.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
+# Compiling to AVM bytecode
+
+The Puyapy compiler can compile Algorand Python smart contracts into directly to AVM bytecode.
```
Comment (achidlow) [inline docs/lg-compile.md:3]: one too many to's here

File: docs/lg-compile.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
+# Compiling to AVM bytecode
+
+The Puyapy compiler can compile Algorand Python smart contracts into directly to AVM bytecode. 
+Once compiled, this bytecode can be utilized to construct AVM Application Call transactions both on and off chain.
+
+## Outputting AVM bytecode from CLI
+
+The `--output-bytecode` option can be used to generate `.bin` files for smart contracts, producing an approval and clear program for each contract, as well as logic signatures.
```
Comment (achidlow) [inline docs/lg-compile.md:8]:
```suggestion
The `--output-bytecode` option can be used to generate `.bin` files for smart contracts and logic signatures, producing an approval and clear program for each smart contract.
```

File: docs/lg-compile.md (hunk 1 above)
Comment (daniel-makerx) [reply docs/lg-compile.md]: two to's is too many

File: docs/lg-compile.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+```{note}
+The Puya compiler incorporates several optimizations that are not present in the bytecode output generated by the 
+[`/v2/teal/compile`](https://developer.algorand.org/docs/rest-apis/algod/#post-v2tealcompile) endpoint.
+When comparing the outputs of Puya and Algod these differences may be observed. 
+To disable these optimizations and produce bytecode identical to `algod` use the `--match-algod-bytecode` option.  
+```
+
+## Obtaining bytecode within other contracts
+
+The [`compile_contract`](#algopy.compile_contract) takes an Algorand Python smart contract and returns a [`CompiledContract`](#algopy.CompiledContract),
```
Comment (achidlow) [inline docs/lg-compile.md:19]:
```suggestion
The [`compile_contract`](#algopy.compile_contract) function takes an Algorand Python smart contract class and returns a [`CompiledContract`](#algopy.CompiledContract),
```

File: docs/lg-compile.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
+# Compiling to AVM bytecode
+
+The Puyapy compiler can compile Algorand Python smart contracts directly into AVM bytecode.
```
Comment (achidlow) [inline docs/lg-compile.md:3]:
```suggestion
The PuyaPy compiler can compile Algorand Python smart contracts directly into AVM bytecode. 
```

**Review by achidlow — APPROVED** [review]

**Pull request merged**